documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc

```diff
@@ -574,6 +586,339 @@ And finally for the time based paging:
+          "$ref" : "#/components/messages/GetSpeedInput" // <1>
+        },
+        "GetSpeedOutput" : { // <2>
+          "$ref" : "#/components/messages/GetSpeedOutput" // <2>
+        }
+      }
+    }
+  },
+  "operations" : {
+    "GetSpeedInput" : {
+      "action" : "receive", // <5>
+      "channel" : {
+        "$ref" : "#/channels/Movement"
+      },
+      "messages" : [ {
+        "$ref" : "#/channels/Movement/messages/GetSpeedInput" // <1>
+      } ]
+    },
+    "GetSpeedOutput" : { // <2>
+      "action" : "send", // <5>
+      "channel" : {
+        "$ref" : "#/channels/Movement"
+      },
+      "messages" : [ {
+        "$ref" : "#/channels/Movement/messages/GetSpeedOutput" // <2>
+      } ]
+    }
+  },
+  "components" : {
+    "messages" : {
+      "GetSpeedInput" : { // <1> <4>
+        "name" : "GetSpeedInput", // <1>
+        "title" : "getSpeedInputPrefferedName",
+        "summary" : "getSpeedInput desc",
+        "content-type" : "application/json",
+        "payload" : {
+          "$ref" : "#/components/schemas/GetSpeedInput" // <1>
+        }
+      },
+      "GetSpeedOutput" : { // <2> <4>
+        "name" : "GetSpeedOutput", // <2>
+        "title" : "GetSpeedOutputPrefferedName",
+        "summary" : "GetSpeedOutput desc",
+        "content-type" : "application/json",
+        "payload" : {
+          "$ref" : "#/components/schemas/GetSpeedOutput" // <2>
+        }
+      }
+    },
+    "schemas" : {
+      "GetSpeedInput" : { // <1> <3>
+        "type" : "string",
+        "description" : "getSpeedInput desc"
+      },
+      "GetSpeedOutput" : { // <2> <3>
+        "type" : "string",
+        "description" : "GetSpeedOutput desc"
+      }
+    }
+  }
+}
+----
+
+<1> the names of the input
+<2> the names of the output
+<3> the characteristics are generated
+<4> the meta information of input/output
+<5> 'send' (for output) action: map to a publish operation in AsyncAPI, 'receive' (for input) action: map to a subscribe operation
+
+==== Mapping of Aspect's events
+
+This section describe mapping between Aspect Events and AsyncAPI spec.
+If the Aspect also has a non-empty list of events defined, like the one in the following example:
+
+[source,turtle]
+----
+:Movement a samm:Aspect ;
+   samm:preferredName "movement"@en ;
+   samm:description "Aspect for movement information"@en ;
+   samm:properties ( ) ;
+   samm:operations ( ) ;
+   samm:events ( :SpeedUpdateEvent ) .
+
+:SpeedUpdateEvent a samm:Event ;
+   samm:preferredName "speedUpdate"@en ;
+   samm:description "This is event for update speed property"@en ;
+   samm:parameters ( :updatedSpeed :propertyName ) .
+
+:updatedSpeed a samm:Property ;
+   samm:preferredName "updatedSpeedPreferredName"@en ;
+   samm:description "Updated Property Desc."@en ;
+   samm:see <http://example.com/> ;
+   samm:see <http://example.com/me> ;
+   samm:exampleValue "1234567890" ;
+   samm:characteristic samm-c:Text .
+
+:propertyName a samm:Property ;
```

Comment:
I think this is an anti-pattern. We should not model a Property that refers to other Properties by name. Please change this so that the Event updates a specific value instead, even if it is not as generic.